### PR TITLE
feat: add a retry mechanism for API call

### PR DESCRIPTION
**Pull request by cheikhgwane**
Add a retry mechanism for the communication with the API

### CODE REVIEW

**Review by qgerome — CHANGES_REQUESTED** [review]
You need to check the `requests` documentation to implement a retry mechanism. No need to implement something custom. We can do it at this level directly

**Comment by cheikhgwane** [comment]
> You need to check the `requests` documentation to implement a retry mechanism. No need to implement something custom. We can do it at this level directly

I've added a httpClient based on Session and Retry object. I've also used the default status_forcelist. Let me know if I need to add/remove status code from that list

File: openhexa/sdk/utils.py [hunk 1]
```diff
@@ -192,3 +194,40 @@ def read_content(source: typing.Union[str, os.PathLike[str], typing.IO, bytes]):
     finally:
         if hasattr(source, "close"):
             source.close()
+
+
+def retry(*args, **kwargs):
```
Comment (qgerome) [inline openhexa/sdk/utils.py:199]: This can be removed.

File: openhexa/utils/HttpClient.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        backoff_factor=backoff_factor,
+        status_forcelist=status_forcelist,
+    )
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+    return session
+
+
+httpClient = requests_builder()
```
Comment (qgerome) [inline openhexa/utils/HttpClient.py:30]: I would not export this but more the `requests_builder`

File: openhexa/utils/HttpClient.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+"""Custom HttpClient with retry mechanism."""
+
+import requests
+from requests import Session
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+
+
+def requests_builder(
```
Comment (qgerome) [inline openhexa/utils/HttpClient.py:9]:
```suggestion
def create_requests_session(
```

**Review by qgerome — CHANGES_REQUESTED** [review]

File: openhexa/utils/HttpClient.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+from requests import Session
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+
+
+def create_requests_session(
+    retries=3,
+    backoff_factor=0.3,
+    status_forcelist=(500, 502, 504),
+    session=None,
```
Comment (qgerome) [inline openhexa/utils/HttpClient.py:13]: Why do you accept a `session` here ?

**Review by qgerome — CHANGES_REQUESTED** [review]

File: openhexa/utils/HttpClient.py
Comment (qgerome) [inline openhexa/utils/HttpClient.py:1]: This should be renamed "session.py" no ?

**Review by qgerome — CHANGES_REQUESTED** [review]

File: openhexa/utils/HttpClient.py (hunk 4 above)
Comment (cheikhgwane) [reply openhexa/utils/HttpClient.py:13]: It shouldn't be here thanks!

**Pull request merged**